lib/winapi/findfile.c

```diff
@@ -0,0 +1,129 @@
+#include <string.h>
+#include <assert.h>
+#include <stdbool.h>
+#include <winbase.h>
+#include <xboxkrnl/xboxkrnl.h>
+#include <hal/fileio.h>
+#include <hal/winerror.h>
+#include <fileapi.h>
+
+struct FileInfo {
+    FILE_DIRECTORY_INFORMATION dirInfo;
+    char filename[MAX_PATH];
+};
+
+static void dirtofind (FILE_DIRECTORY_INFORMATION *dirInfo, LPWIN32_FIND_DATAA lpFindFileData)
+{
+    lpFindFileData->dwFileAttributes = dirInfo->FileAttributes;
+    lpFindFileData->ftCreationTime.dwLowDateTime = dirInfo->CreationTime.LowPart;
+    lpFindFileData->ftCreationTime.dwHighDateTime = dirInfo->CreationTime.HighPart;
+    lpFindFileData->ftLastAccessTime.dwLowDateTime  = dirInfo->LastAccessTime.LowPart;
+    lpFindFileData->ftLastAccessTime.dwHighDateTime = dirInfo->LastAccessTime.HighPart;
+    lpFindFileData->ftLastWriteTime.dwLowDateTime  = dirInfo->LastWriteTime.LowPart;
+    lpFindFileData->ftLastWriteTime.dwHighDateTime = dirInfo->LastWriteTime.HighPart;
+    lpFindFileData->nFileSizeHigh = dirInfo->EndOfFile.HighPart;
+    lpFindFileData->nFileSizeLow = dirInfo->EndOfFile.LowPart;
+    memcpy(lpFindFileData->cFileName, dirInfo->FileName, dirInfo->FileNameLength);
+    lpFindFileData->cFileName[dirInfo->FileNameLength] = '\0';
```

Comment:
Is it possible for `dirInfo->FileNameLength == MAX_PATH` potentially overrunning `cFileName` (buffer of size `MAX_PATH`) or will it always be strictly `< MAX_PATH`?

Reply:
It was possible, but I just fixed it. The buffer provided by `struct FileInfo` is now `MAX_PATH-1`, which makes sure we don't overflow the output buffer.
I had a similar error in the assertion, which should check `< MAX_PATH` and not `<= MAX_PATH`.

Reply:
I was very confused by this comment at first, but it all makes sense. Thanks :+1: